### DISCUSSION TITLE
Generate annotations on model.

### DISCUSSION THREAD

**Discussion by fmg-kapil** [asyncapi/modelina#1888] [Q&A]
Is there a method to generate annotations on the models in asyncapi.yaml file or any other means ? I need a model with annotations to be generated: 

[BindingKey("MiningBlock.Archived")]
[RoutingKey("MiningBlock.Archived")]
public partial class MiningBlockArchivedEvent
 {
      public string? Id { get; set; }
      public string? FullName { get; set; }
      public string? Timestamp { get; set; }
      public Dictionary<string, dynamic>? AdditionalProperties { get; set; }
 }

Please let me know if this is possible ? This is very urgent piece for my work to progress. 

Thanks in advance

**Comment by jonaslagoni** [comment]
@fmg-kapil have a look at [Presets](https://modelina.org/docs/presets) :v:

You can use CSHARP_JSON_SERIALIZER_PRESET that also adds annotation to the class as reference https://github.com/asyncapi/modelina/blob/ff2d0cf54b53c3013e0e79013aa8f6cf4d8552d6/src/generators/csharp/presets/JsonSerializerPreset.ts#L220

**Reply by fmg-kapil** [reply]
Yes i observed that yesterday and ran this example : npm i && npm run start:windows in the folder : /async-api/modelina/examples/csharp-generate-json-serializer but I want to customize this annotation as I described in my original question on 13.Mar.  
a) Do I then need to make my own preset and submit a pull request ? 
b) how to use it from the command line then as I want to embed this command of generating models [ with annotations] in the .csproj file. I know there is a switch :--csharpSystemJson but how do I use my own preset ?

Please reply so I can take it up urgently.

**Reply by jonaslagoni** [reply]
> a) Do I then need to make my own preset and submit a pull request ?

That would definitely be the preferred approach, cause then others can benefit from the preset as well 👍 If the annotations are something others can use of course.

> b) how to use it from the command line then as I want to embed this command of generating models [ with annotations] in the .csproj file. I know there is a switch :--csharpSystemJson but how do I use my own preset ?

Yea, it's a problem right now that the CLI does not support custom presets... You would have to setup a small script that calls Modelina explicitly instead of through the CLI at least until https://github.com/asyncapi/modelina/issues/780 and https://github.com/asyncapi/modelina/issues/1724 is solved.

**Reply by fmg-kapil** [reply]
Thanks very much for your reply. I did make changes to CSHARP_JSON_SERIALIZER_PRESET on my local copy of the repo and the results were as I expected.  I think this change will require changes to both the cli and modelina repos. 

In cli repo :  (somthink along these lines) 
if (csharpGenerateAnnotations) {
        presets.push(CSHARP_ANNOTATIONS_PRESET);
      }

In modelina repo: 
- make a new preset : CSHARP_ANNOTATIONS_PRESET

I will then run both these as local docker images on my machine and see if everything hooks up together. It may be a bit more complex than it seems. 

Regards,
